### PR TITLE
Address customer feedback, add basic consumption pattern

### LINKED ISSUE #26769: application settings app.config / appsettings.json console and desktop apps
what kind of crazy drugs were used ?
with .net we had the xml app.config file and that has worked fine.

and i am ok with moving to json but what i am not ok with is now the settings has turned into a huge mess.....
i dont want to spend all day just to figure out how to do this!   
so why even go to the new json?
and how ?
and i find so many links that assume i need to do it in a web app.... what about desktop and console apps ?

this is one area that the new .net 6  and 5 and core have just made a mess of things....

can we please have a simple way to do settings for non web apps ?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b458a209-339d-e896-4e85-6d1c6a7aa1f9
* Version Independent ID: 0d5de37c-8f89-b6a5-3132-2c5214196bfd
* Content: [Configuration in .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/configuration)
* Content Source: [docs/core/extensions/configuration.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/configuration.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### LINKED ISSUE #27169: The "Configure console apps" section doesn't actually say how to use the config
Do I need to use `host.Services.GetRequiredService` to get the configuration service? The documentation doesn't explain.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b458a209-339d-e896-4e85-6d1c6a7aa1f9
* Version Independent ID: 0d5de37c-8f89-b6a5-3132-2c5214196bfd
* Content: [Configuration in .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/configuration)
* Content Source: [docs/core/extensions/configuration.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/configuration.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Address customer feedback, add basic consumption pattern for configuration.

Fixes #27169 and fixes #26769

✔️ [Internal Preview: Configuration in .NET](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/configuration?branch=pr-en-us-27179)

### CODE REVIEW

File: docs/core/extensions/snippets/configuration/console-basic/Program.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+// Get values from the config given their key and their target type.
+int keyOneValue = config.GetValue<int>("KeyOne");
+bool keyTwoValue = config.GetValue<bool>("KeyTwo");
+string keyThreeNestedValue = config.GetValue<string>("KeyThree:Message");
+
+// Write the values to the console.
+Console.WriteLine($"KeyOne = {keyOneValue}");
+Console.WriteLine($"KeyTwo = {keyTwoValue}");
+Console.WriteLine($"KeyThree:Message = {keyThreeNestedValue}");
```
Comment (Daniel15) [inline docs/core/extensions/snippets/configuration/console-basic/Program.cs:18]: Is it worth adding an example that binds to an object? I thought I'd need an `IOptions<T>` for that, but this works fine, I guess as long as `Microsoft.Extensions.Configuration.Binder` is installed.
```
public struct MyAppConfig {
    private string Name { get; init; }
}
...
config.GetValue<MyAppConfig>("MyApp")
```

File: docs/core/extensions/configuration.md [hunk 2]
```diff
@@ -39,7 +40,38 @@ Configuration providers that are added later override previous key settings. For
 
 ### Binding
 
-One of the key advantages to configuration in .NET is the ability to bind configuration values to instances of .NET objects. For example, the JSON configuration provider can be used to map *appsettings.json* files to .NET objects and is used with dependency injection. This enables the options pattern, the options pattern uses classes to provide strongly typed access to groups of related settings.
+One of the key advantages of using the .NET configuration abstractions is the ability to bind configuration values to instances of .NET objects. For example, the JSON configuration provider can be used to map *appsettings.json* files to .NET objects and is used with [dependency injection](dependency-injection.md). This enables the [options pattern](options.md), the options pattern uses classes to provide strongly typed access to groups of related settings. .NET configuration provides various abstractions, consider the following interfaces:
```
Comment (tdykstra) [inline docs/core/extensions/configuration.md:43]:
```suggestion
One of the key advantages of using the .NET configuration abstractions is the ability to bind configuration values to instances of .NET objects. For example, the JSON configuration provider can be used to map *appsettings.json* files to .NET objects and is used with [dependency injection](dependency-injection.md). This enables the [options pattern](options.md), which uses classes to provide strongly typed access to groups of related settings. .NET configuration provides various abstractions. Consider the following interfaces:
```

File: docs/core/extensions/configuration.md [hunk 3]
```diff
@@ -39,7 +40,38 @@ Configuration providers that are added later override previous key settings. For
...
+### Basic example
+
+To access the `IConfiguration` value, you can rely again on the [`Microsoft.Extensions.Hosting`](https://www.nuget.org/packages/Microsoft.Extensions.Hosting) NuGet package. Create a new console application, and paste the following project file contents into it:
+
+:::code language="xml" source="snippets/configuration/console-basic/console-basic.csproj" highlight="4,11-13,17":::
+
+The preceding project file defines:
+
+- That the application is an executable.
+- That an _appsettings.json_ file is to be copied to the output directory when compiled.
```
Comment (tdykstra) [inline docs/core/extensions/configuration.md:60]:
```suggestion
- That an _appsettings.json_ file is to be copied to the output directory when the project is compiled.
```

File: docs/core/extensions/configuration.md [hunk 4]
```diff
@@ -39,7 +40,38 @@ Configuration providers that are added later override previous key settings. For
...
+
+Add the _appsettings.json_ file at the root of the project with the following contents:
+
+:::code language="json" source="snippets/configuration/console-basic/appsettings.json":::
+
+Replace the contents of the _Program.cs_ file with the following C# code:
+
+:::code language="csharp" source="snippets/configuration/console-basic/Program.cs" highlight="5,8,11-13,16-18":::
+
+When you run this application, the `Host.CreateDefaultBuilder` defines the behavior to discover the JSON configuration and expose it through the `IConfiguration`. From the `host` instance, you can ask the service provider for the `IConfiguration` instance and then ask it for values.
```
Comment (tdykstra) [inline docs/core/extensions/configuration.md:71]:
```suggestion
When you run this application, the `Host.CreateDefaultBuilder` defines the behavior to discover the JSON configuration and expose it through the `IConfiguration` instance. From the `host` instance, you can ask the service provider for the `IConfiguration` instance and then ask it for values.
```

File: docs/core/extensions/configuration.md [hunk 5]
```diff
@@ -39,7 +40,38 @@ Configuration providers that are added later override previous key settings. For
...
+
+> [!TIP]
+> Using the raw `IConfiguration` in this way, while convenient, doesn't scale very well. When applications grow in complexity, and their corresponding configurations become more sophisticated, it's recommended to use the [_options pattern_](options.md) as an alternative.
```
Comment (tdykstra) [inline docs/core/extensions/configuration.md:74]:
```suggestion
> Using the raw `IConfiguration` instance in this way, while convenient, doesn't scale very well. When applications grow in complexity, and their corresponding configurations become more complex, we recommend that you use the [_options pattern_](options.md) as an alternative.
```

**Review by tdykstra — APPROVED** [review]
Looks good, a couple suggestions.

File: docs/core/extensions/snippets/configuration/console-basic/Program.cs (hunk 1 above)
Comment (IEvangelist) [reply docs/core/extensions/snippets/configuration/console-basic/Program.cs:18]: @Daniel15  What does all of your code look like? I'll add this but need to test it locally.

**Pull request merged**